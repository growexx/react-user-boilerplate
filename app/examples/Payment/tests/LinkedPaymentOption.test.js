import React from 'react';
import { render } from 'react-testing-library';
import LinkedPaymentOption from '../LinkedPaymentOption';

const props = {};

const componentWrapper = updatedProps =>
  render(<LinkedPaymentOption {...props} {...updatedProps} />);

describe('<LinkedPaymentOption/>', () => {
  it('should render with no props', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with props type paypal', () => {
    const {
      container: { firstChild },
    } = componentWrapper({
      type: 'Paypal',
    });
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with props type paytm', () => {
    const {
      container: { firstChild },
    } = componentWrapper({
      type: 'Paytm',
    });
    expect(firstChild).toMatchSnapshot();
  });
});
