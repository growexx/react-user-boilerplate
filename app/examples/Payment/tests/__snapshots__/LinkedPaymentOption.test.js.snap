// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkedPaymentOption/> should render with no props 1`] = `null`;

exports[`<LinkedPaymentOption/> should render with props type paypal 1`] = `
<div
  class="ant-card common-payment-main"
>
  <div
    class="ant-card-body"
  >
    <div
      class="payment-main-card"
    >
      <span
        class="ant-input-affix-wrapper"
      >
        <input
          class="ant-input"
          placeholder="Enter Paypal Linked No."
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <button
            class="ant-btn ant-btn-link"
            disabled=""
            type="button"
          >
            <span>
              Link
            </span>
          </button>
        </span>
      </span>
      <button
        class="ant-btn ant-btn-primary continue-btn"
        disabled=""
        type="button"
      >
        <span>
          Continue
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<LinkedPaymentOption/> should render with props type paytm 1`] = `
<div
  class="ant-card common-payment-main"
>
  <div
    class="ant-card-body"
  >
    <div
      class="payment-main-card"
    >
      <span
        class="ant-input-affix-wrapper"
      >
        <input
          class="ant-input"
          placeholder="Enter Paytm Linked No."
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <button
            class="ant-btn ant-btn-link"
            disabled=""
            type="button"
          >
            <span>
              Link
            </span>
          </button>
        </span>
      </span>
      <button
        class="ant-btn ant-btn-primary continue-btn"
        disabled=""
        type="button"
      >
        <span>
          Continue
        </span>
      </button>
    </div>
  </div>
</div>
`;
